src/components/App/OptionDropdown.vue

```diff
@@ -0,0 +1,199 @@
+        default: false,
+        type: Boolean
+    }
+  },
+  computed: {
+    defaultOptionMargin() {
+        // default margin-top (in pixels) of the option dropdown without any scrolling effect
+        return this.convertRemToPixels(this.defaultOptionMarginRem)
+    },
+    dropdownIconClass() {
+        // class for the dropdown icon
+      return [
+        { "transform rotate-180": this.showDropdown },
+        "transition ease duration-800 text-gray-600",
+      ];
+    },
+  },
+  watch: {
+    defaultOptionMarginRem() {
+        // update option box margin when the default top margin is updated
+        this.setOptionBoxStyling()
+    },
+    scrollY() {
+        // update option box margin when the page is scrolled is updated
+        this.setOptionBoxStyling()
+    },
+    showDropdown() {
+        // if the dropdown is closed, reset the option box's margin-top
+        if (!this.showDropdown) {
+            this.defaultOptionMarginRem = 2
+        }
+    }
+  },
+  created() {
+      // set the option box margin when the page is created
+      this.setOptionBoxStyling()
+  },
+  methods: {
+    ...Utilties,
+    optionClass(index) {
+        // class for each option
+        return {
+            'opacity-50 cursor-not-allowed': this.isOptionDisabled(index),
+            'hover:bg-primary hover:cursor-pointer hover:text-white': !this.isOptionDisabled(index)
+        }
+    },
+    isOptionDisabled(index) {
+        // whether the option at the given index is disabled
+        return this.options[index].disabled
+    },
+    setOptionBoxStyling() {
+        // sets the option box margin based on various parameters
+        if (!this.scrollY) {
+            this.optionBoxStyling = { 'margin-top': `${this.defaultOptionMarginRem}rem` }
+        } else {
+            this.optionBoxStyling = { 'margin-top': `-${this.scrollY - this.defaultOptionMargin}px`}
+        }
+    },
+    convertRemToPixels(rem) {
+        // converts value in rem to pixels
+        return rem * parseFloat(getComputedStyle(document.documentElement).fontSize);
+    },
+    doesOptionHaveIcon(option) {
+      // whether the given option has an icon
+      return option != undefined && option.icon != null;
+    },
+    hideDropdown() {
+      // hides the dropdown when clicking away from the dropdown
+      this.showDropdown = false;
+      this.$emit("toggle-visibility", this.showDropdown);
+    },
+    setOption(index) {
+      // sets the given index as the selected option index
+      if (this.isOptionDisabled(index)) return
+      this.toggleDropdownDisplay();
+      this.$emit('select', index, this.options[index].value)
+    },
+    isOptionsOverflowBottom() {
+        // checks whether the bottom of the options overflows the screen
+        const optionsContainer = document.querySelector('#options')
+        const rectangle = optionsContainer.getBoundingClientRect()
+        return rectangle.bottom > window.innerHeight
```

Comment:
```suggestion
        return rectangle.bottom >= window.innerHeight
```
We don't want the option box to stick to the lower bottom as well